middle_end/flambda2/to_cmm/to_cmm_set_of_closures.ml

```diff
@@ -0,0 +1,431 @@
+(**************************************************************************)
+(*                                                                        *)
+(*                                 OCaml                                  *)
+(*                                                                        *)
+(*                        Guillaume Bury, OCamlPro                        *)
+(*                                                                        *)
+(*   Copyright 2019--2022 OCamlPro SAS                                    *)
+(*   Copyright 2022 Jane Street Group LLC                                 *)
+(*                                                                        *)
+(*   All rights reserved.  This file is distributed under the terms of    *)
+(*   the GNU Lesser General Public License version 2.1, with the          *)
+(*   special exception on linking described in the file LICENSE.          *)
+(*                                                                        *)
+(**************************************************************************)
+
+[@@@ocaml.warning "+a-30-40-41-42"]
+
+open! Flambda.Import
+module Env = To_cmm_env
+module Ece = Effects_and_coeffects
+module R = To_cmm_result
+
+module C = struct
+  include Cmm_helpers
+  include To_cmm_shared
+end
+
+type translate_expr =
+  To_cmm_env.t -> To_cmm_result.t -> Expr.t -> Cmm.expression * To_cmm_result.t
+
+(* Filling of closure blocks *)
+
+let get_whole_closure_symbol =
+  let whole_closure_symb_count = ref 0 in
+  fun ~set_of_closures_symbol_ref ->
+    match !set_of_closures_symbol_ref with
+    | Some set_of_closures_symbol -> set_of_closures_symbol
+    | None ->
+      incr whole_closure_symb_count;
+      let comp_unit = Compilation_unit.get_current_exn () in
+      let linkage_name =
+        Linkage_name.create
+          (Printf.sprintf ".clos_%d" !whole_closure_symb_count)
+      in
+      let set_of_closures_symbol = Symbol.create comp_unit linkage_name in
+      set_of_closures_symbol_ref := Some set_of_closures_symbol;
+      set_of_closures_symbol
+
+module Make_layout_filler (P : sig
+  type cmm_term
+
+  val int : dbg:Debuginfo.t -> nativeint -> cmm_term
+
+  val simple :
+    To_cmm_env.t ->
+    Simple.t ->
+    [`Data of cmm_term list | `Var of Variable.t] * To_cmm_env.t * Ece.t
+
+  val infix_header : dbg:Debuginfo.t -> first_value_slot_offset:int -> cmm_term
```

Comment:
The name of the argument `first_value_slot_offset` is very confusing to me... What is actually given to the infix header is simply the offset of the closure inside the set of closures. The current name suggests (at least to me) a relation to the start of env, whereas there is none, so could you change the name ?

Reply:
Sorry, yes that isn't right.  Maybe it should be `function_slot_offset`?